### PR TITLE
Add openstack project module

### PR DESCRIPTION
**Pull request by agireud**
Added os_project module to support openstack projects.

### CODE REVIEW

**Comment by robynbergeron** [comment]
Hi @agireud -- 

Thanks for submitting this new module to Ansible Extras! This module is now in community review, a process that is open to all Ansible users. In order for this module to be approved, it must gain the following votes, which can be contributed by anyone other than the author of the new module:

“works_for_me”: If you have tested the module thoroughly, including testing of all of the module’s options, and if the module works for you, please add “works_for_me” in the comments.

“passes_guidelines”: If you have gone through the module guidelines and the module meets all of the requirements, please add “passes_guidelines” in the comments. Guidelines are available here: http://docs.ansible.com/developing_modules.html#module-checklist

“needs_revision”: If the module fails to work for you, or if it doesn’t meet guidelines, please add “needs_revision” in the comments with details about what needs to be fixed.

When a module has both “works_for_me” and “passes_guidelines” tags, we will promote the module for inclusion in Ansible Extras.  At this point, you will be expected to maintain the module by fixing bugs and evaluating pull requests in a timely manner. 

More information about the review process can be seen here: 
https://github.com/ansible/ansible-modules-extras/blob/devel/REVIEWERS.md

Thanks again for submitting your Ansible module!

**Comment by robynbergeron** [comment]
@agireud -- In addition to the above guidelines, since this is a new openstack module, also looking for commentary / feedback from the existing maintainers of the openstack modules in ansible. 

Pinging @shrews @dguerri @juliakreger @j2sol @rcarrillocruz @emonty for that feedback / commentary :D

**Comment by emonty** [comment]
Hey - this looks generally great, thanks for helping! - but it should really be made against ansible/ansible-modules-core, not here.

**Comment by agireud** [comment]
Not a problem, happy to help!  @emonty, that was my original thought since all the other OpenStack modules are located there but after inquiring in IRC I was told to open it in extras.  The module contribution guidelines also mention that any new modules should go in extras.

I'll go ahead and close this then resubmit the pull request to ansible/ansible-modules-core.

**Comment by Shrews** [comment]
@agireud Also, go ahead and add support for check mode. You have everything there to easily add it.

**Comment by agireud** [comment]
@Shrews I went ahead and added check mode in the other PR.  Thanks for the input.

**Comment by agireud** [comment]
@bcoca, @emonty, I'm getting conflicting instructions from the both of you.  My pull request to ansible-modules-core was closed and I was told that it should be submitted here instead.  

https://github.com/ansible/ansible-modules-core/pull/2433

**Comment by omgjlk** [comment]
Sorry about that. This spawned a convo on the mailing list, to try and
resolve this so that you (and future people) don't get the run-around.
- jlk

**Comment by sivel** [comment]
Yes, the consensus is that the PR should be opened here.  I'll re-open this PR.

**Comment by agireud** [comment]
Thanks everyone, sorry for the confusion.

I went ahead and added support for check mode to this branch.  Should be ready for review now.

File: cloud/openstack/os_project.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    type: dictionary
+    contains:
+        description:
+            description: Project description
+            type: string
+            sample: "demodescription"
+        domain_id:
+            description: Project domain ID. Only present with Keystone >= v3.
+            type: string
+            sample: "default"
```
Comment (omgjlk) [inline cloud/openstack/os_project.py:98]: Would this be the domain ID (uuid) or the name "default" that you list here?

**Comment by omgjlk** [comment]
passes_guidelines
works_for_me

File: cloud/openstack/os_project.py (hunk 1 above)
Comment (agireud) [inline cloud/openstack/os_project.py:98]: I intended it to be Domain ID (uuid) but it looks like I might have been mistaken in including it here based off of https://github.com/openstack/python-keystoneclient/blob/master/keystoneclient/v3/projects.py#L26.

**Pull request merged**